lib/actions/document_ai.ex

```diff
@@ -0,0 +1,98 @@
+defmodule WorkflowEngine.Actions.DocumentAi do
+  @doc """
+  Executes a request against the Azure Document AI API.
+  """
+
+  use OK.Pipe
+
+  @behaviour WorkflowEngine.Action
+
+  require Logger
+
+  @default_document_ai_api_version "2023-07-31"
+
+  def execute(state, %{"type" => "document_ai"} = step) do
+    with {:ok, model_id} <- get_property(step, "model_id"),
+         {:ok, document_url} <- get_property(step, "document_url"),
```

Comment:
Btw, in the ingest_shipment_data action (a workflow engine action private to the Scanpage), we support both static values and paths to extract a value out of (instead of full JSONLogic), e.g.: https://github.com/box-id/service_scan_page_v2/blob/1080ed7b619a4307e307597eb5e8cdd1f216e3be/lib/scanpage_v2/workflow_engine/actions/ingest_shipment_data.ex#L11-L12

Implementation: https://github.com/box-id/service_scan_page_v2/blob/1080ed7b619a4307e307597eb5e8cdd1f216e3be/lib/scanpage_v2/workflow_engine/actions/ingest_shipment_data.ex#L92-L124

But I'm not sure if this is the way to go 100%. Open for feedback